modules/nf-core/cafe/main.nf

```diff
@@ -0,0 +1,56 @@
+process CAFE {
+    label 'process_high'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/cafe:5.1.0--h43eeafb_0':
+        'biocontainers/cafe:5.1.0--h43eeafb_0' }"
+
+    input:
+    path(infile)
+    path(tree)
+
+    output:
+    path("Out_folder")     , emit: cafe
+    path("versions.yml")   , emit: versions
+    // I want to add the below out paths, but when I do this. nf-test fails for snapshot. We cannot snapshot the entire out folder, as it has time stamps in files and random numbers
+    //path("Out_folder/Base_count.tab") , emit: cafe_base_count
+    //path("Out_folder/*.tre") , emit: cafe_significant_trees
+    //path("Out_folder/*_report.cafe") , emit: cafereport
+    //path("Out_folder/*results.txt") , emit: cafe_results
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def VERSION = '5.1.0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    """
+    tr '\\r' '\\n' < $infile > infile.txt
+    tr '\\r' '\\n' < $tree > treefile.txt
+    cafe5 \\
+        -i infile.txt \\
+        -t treefile.txt \\
+        $args \\
+        --cores ${task.cpus} \\
+        -o Out_folder
```

Comment:
Why is this not just the work directory itself?
```suggestion
        -o .
```

Reply:
OK, yes it can be. I was not sure whether this should be a user chosen variable.

Like: https://github.com/nf-core/modules/blob/master/modules/nf-core/agat/spstatistics/main.nf

`--output ${prefix}.stats.txt `

So that we can find all the results folder in a specifically named directory.. But maybe that will work too with `-o  .`

I will try it

Reply:
Cool, ok thanks, 
I tried it and it does work fine that way, and now I think the broken snapshots are working.
Thanks for the help by the way, I haven't contributed many modules, so am a little unsure how best to do it

Reply:
Ah, again I get the same issue. 

I will write on slack as well. 

When I have a single out file emitted:
```
path(".")     , emit: cafe
path("versions.yml")   , emit: versions
path("Base_count.tab") , emit: cafe_base_count
```

```
│   Test [eb9ce4a5] 'mycoplasma - genecounts - tree' PASSED (27.586s)                                                                                                                                         │
│   Test [8d9c689b] 'mycoplasma - genecounts - tree - stub' PASSED (4.54s)                                                                                                                                    │
│                                                                                                                                                                                                             │
│                                                                                                                                                                                                             │
│ SUCCESS: Executed 2 tests in 32.135s     
```

The nf-test works . Albeit without the snapshot for now. 

Then when I add additional emit statements it fails:
```
path(".")     , emit: cafe
path("versions.yml")   , emit: versions
path("Base_count.tab") , emit: cafe_base_count
path("Base_asr.tre") , emit: cafe_significant_trees
```

ERROR:
```
│   Assertion failed:                                                                                                                                                                                         │
│                                                                                                                                                                                                             │
│   1 of 1 assertions failed                                                                                                                                                                                  │
│                                                                                                                                                                                                             │
│   Nextflow stdout:                                                                                                                                                                                          │
│                                                                                                                                                                                                             │
│   ERROR ~ Error executing process > 'CAFE (1)'                                                                                                                                                              │
│                                                                                                                                                                                                             │
│   Caused by:                                                                                                                                                                                                │
│     Missing output file(s) `Base_asr.tre` expected by process `CAFE (1)`                                                                                                                                    │
│                                                                                                                                                                                                             │
│                                                                                                                                                                                                             │
│   Command executed:                                                                                                                                                                                         │
│                                                                                                                                                                                                             │
│     touch Base_count.tab                                                                                                                                                                                    │
│                                                                                                                                                                                                             │
│     cat <<-END_VERSIONS > versions.yml                                                                                                                                                                      │
│     "CAFE":                                                                                                                                                                                                 │
│         cafe: 5.1.0                                                                                                                                                                                         │
│     END_VERSIONS                                                                                                                                                                                            │
│                                                                                                                                                                                                             │
│   Command exit status:                                                                                                                                                                                      │
│     0                                                                                                                                                                                                       │
│                                                                                                                                                                                                             │
│   Command output:                                                                                                                                                                                           │
│     (empty)                                                                                                                                                                                                 │
│                                                                                                                                                                                                             │
│   Work dir:                                                                                                                                                                                                 │
│     /Users/cwyatt/Desktop/modules/.nf-test/tests/8d9c689bcc34045023ae209756425d5b/work/e1/30910eee61c74fc3b92f1af9cc9a54                                                                                    │
│                                                                                                                                                                                                             │
│   Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`                                                                                         │
│                                                                                                                                                                                                             │
│    -- Check '/Users/cwyatt/Desktop/modules/.nf-test/tests/8d9c689bcc34045023ae209756425d5b/meta/nextflow.log' file for details                                                                              │
│   Nextflow stderr:    
```

Reply:
That one is complaining that your stub test is not making the tre file:
```
Command executed:                                                                                                                                                                                         
                                                                                                                                                                                                             
touch Base_count.tab   
```
So you'd need to also touch the .tre file in the stub test.

Reply:
Ahhh ok. That makes sense, I hadn't appreciated what the stub test was doing. I will try to add that in

Reply:
Thanks @SPPearce . I was missing a few commas, but thanks to your help, I think it is working as expected now. Just have some linting issues, and probably I need to put the raw data onto nf-core too, instead of our private S3

Reply:
Not sure now why the `nf-core-lint-modules (cafe) (pull_request)` fails. It says versions is not in the snapshot, but not sure why it is expecting it to be there. I only only assert a snapshot on a time that doesn't have a time stamp, though could add versions to the snapshot too (as this should not change)

![image](https://github.com/user-attachments/assets/0e3cfd7a-44f5-4eb6-837c-d70c620b5030)

Many other run tests on other modules fail too, and am not sure why